text/0000-pack-publish-buildpack.md

```diff
@@ -0,0 +1,176 @@
+    - Close the Github issue.
+
+**NOTE:** Github Issues that are not recognized as a request to add/yank a buildpack version will be automatically closed. All PRs will also be automatically closed.
+
+## Headless: `pack publish-buildpack --token <token> <url>`
+
+When the `pack publish-buildpack` command is passed the `--token` option, it will use an API to publish the buildpack. However, it will require an API token from the user.
+
+Arguments:
+
+* `url` - the location of a buildpackage _in an Docker registry_ to be released in the buildpack registry
+
+Steps:
+
+1. Pulls the image manifest for the buildpackage at `url` if it does not already exist locally.
+1. Reads the following information from the image manifest (if any of these are missing, the command fails):
+    - `id` (must include a `namespace` and `name` in the form `<namespace>/<name>`)
+    - `version`
+1. Makes an HTTP POST request to the endpoint `<registry-write-url>/<namespace>/<name>@<version>` with the following:
+    - Header: `Authorization: Bearer <token>`
+    - Body: a JSON payload represeting the buildpack version as defined in [RFC-0022](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md)
+
+Option:
+
+* `--token <token>` - an authorization token for the remote registry service
+
+**NOTE:** This command requires the default Registry URL be split into a read URL (for the Git pull) and a write URL (for the HTTP POST). Alternatively, we could define a specification for resolving a Git URL from a unified registry URL.
+
+## Native: `pack publish-buildpack --native <url>`
+
+When the `pack publish-buildpack` command is passed the `--native` option, it will commit the buildpack directly to the local registry cache, and attempt to push to the origin. This will NOT work against the official registry index, and the CLI will explicitly check for this before pushing so that it can error out with an appropriate message.
+
+Arguments/Options:
+
+* `url` - the location of a buildpackage _in an Docker registry_ to be released in the buildpack registry
+
+Steps:
+
+1. Pulls the image manifest for the buildpackage at `url` if it does not already exist locally.
+1. Reads the following information from the image manifest (if any of these are missing, the command fails):
+    - `id` (must include a `namespace` and `name` in the form `<namespace>/<name>`)
+    - `version`
+1. Makes a Git commit to the local registry index with a JSON payload represeting the buildpack version as defined in [RFC-0022](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md)
+1. Uses Git to push the commit to the registry index's `origin`.
+
+Option:
+
+* `--native` - indicates that the native Git mechanism should be used
+
+## `pack yank-buildpack <buildpack-id-and-version>`
```

Comment:
Will there be a corresponding `--native` version of this command?

Reply:
yea, it probably should. we're going to revise this a bit, so I'll be sure to include that